addons/mrp/models/stock_orderpoint.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class StockWarehouseOrderpoint(models.Model):
+    _inherit = 'stock.warehouse.orderpoint'
+
+    show_bom = fields.Boolean('Show BoM column', compute='_compute_show_bom')
+    bom_id = fields.Many2one(
+        'mrp.bom', string='Bill of Materials',
+        domain="[('type', '=', 'normal'), '|', ('product_id', '=', product_id), '&', ('product_id', '=', False), ('product_tmpl_id', '=', product_tmpl_id)]")
+
+    def _compute_show_bom(self):
+        for orderpoint in self:
+            orderpoint.show_bom = any(ru.action == 'manufacture' for ru in orderpoint.route_id.rule_ids)
+
+    def _get_default_route_id(self):
```

Comment:
default should be declare dbefore compute

Reply:
not a real default so you can disregard this comment
but! you should really batch this method.. and it looks similar to `_compute_show_bom` in the lookup for manufacture rule. so maybe a chance to share some code ?